test/logger.test.js

```diff
@@ -191,14 +195,52 @@ describe('logger', () => {
             });
             sinon.spy(logger, '_createBulk');
 
+            logger.log('hello there from test');
+            assert.equal(logger._createBulk.getCall(0).args[0][0].hasOwnProperty('@timestamp_nano'), true);
+
+            logger._createBulk.restore();
+
+            //testing length of nanosec-timestamp
+            const nanosec_index = 3;
+            const mock_log = {
+                message: 'hello there from test', 
+            };
+            logger._addTimestamp(mock_log);
+            let mock_nano_timestamp = mock_log['@timestamp_nano'].split("-")[nanosec_index];
+
+            assert.equal(mock_nano_timestamp.length, nanoSecDigits);
+
+            logger.close();
+        });
+
+        it('pad nano-sec timestamp with zeros', (done) => {
```

Comment:
Please merge both tests.
What the test should do is to expect the timestamp to be something you'd know (like `expectedTimestamp`) and then see that that was indeed the timestamp of the log.
You'd probably have to create relevant mocks.

Reply:
Divided to 3 tests:
default case of no nanosec-timestamp
add nano-sec timestamp option
valid nano-sec timestamp
c326c92
@AmirTugi